src/mapping.ts

```diff
@@ -9,170 +9,158 @@ import {
-//  */
-// function decodeValue(nftValue: BigInt): Values {
-//   let value: Values;
-//   let _tier: BigInt = nftValue / CLASS_MULTIPLIER;
-//   let tier = checkMergeClass(_tier);
-
-//   value = {
-//     mass: (nftValue % CLASS_MULTIPLIER) as BigInt,
-//     tier: tier,
-//     color: checkColor(tier),
-//   };
-
-//   return value;
-// }
+import { User, NFT, Collection } from "../generated/schema";
+import {
+  EMPTY_ADDRESS,
+  CLASS_MULTIPLIER,
+  NG_OMNIBUS,
+  MERGE_EXTADDR,
+  MERGE_NAME,
+  TOKENSTANDARD,
+  BIGINT_ZERO,
+  ADDR_DEAD,
+} from "./utils";
 
 /* ========== EVENT HANDLERS ========== */
 
 /**
  * @notice handles Transfer events and ultimately instantiates / updates User and NFT entities
  * @param event: Transfer
  * @dev all NFT entity generation is handled in other functions.
+ * Step 0 && Step 1: set NG_OMNIBUS as a whitelisted address as it is done in the constructor, and initialize new users
+ * Step 2a: either create new NFT if in minting phase
+ * Step 2b: OR updateNFT for all other Transfer scenarios incl: temporary transfers right before merging/burning, simple burning (direct to dead address, which then transfers to address(0))
+ * NOTE: MassUpdates will cover the merge scenario.
  */
 export function handleTransfer(event: Transfer): void {
   const to = event.params.to.toHex();
   const from = event.params.from.toHex();
 
   let user = User.load(to);
   const tokenId = event.params.tokenId.toString();
+  let collection = Collection.load(MERGE_EXTADDR);
+
+  if (!collection) {
+    collection = createCollection(MERGE_EXTADDR); // NEW for collection entity - first instance of minting
+  }
 
-  // Step 0 && Step 1: set NG_OMNIBUS as a whitelisted address as it is done in the constructor, and initialize new users
   if (!user) {
     user = new User(to);
-    if (NG_OMNIBUS) {
+    collection.totalUniqueOwnerAddresses++;
+    collection.save();
+    user.save();
+    if (to == NG_OMNIBUS) {
       user.whitelist = true;
     }
   }
 
-  user.save();
-
-  let nft: NFT;
-
-  // Step 2a: either create new NFT if in minting phase
+  let scenario: string;
   if (from == EMPTY_ADDRESS && to == NG_OMNIBUS) {
-    nft = createNFT(event, user.id);
-  } else {
-    // Step 2b: OR updateNFT for all other Transfer scenarios incl: temporary transfers right before merging/burning, simple burning (direct to dead address, which then transfers to address(0))
-    nft = updateNFT(tokenId, user.id);
+    createNFT(event, user.id);
+    scenario = "Mint";
+    updateCollection(tokenId, scenario);
+  } else if (to == EMPTY_ADDRESS && from == NG_OMNIBUS) {
+    scenario = "NiftyBurn";
+    updateNFT(tokenId, user.id);
+    updateCollection(tokenId, scenario);
+  } else if (to == EMPTY_ADDRESS) {
+    scenario = "NonNiftyBurn";
+    updateNFT(tokenId, user.id);
+    updateCollection(tokenId, scenario);
+  } else if (from == ADDR_DEAD) {
```

Comment:
what made you use `ADDR_DEAD`?

Also, if you are burning a token, shouldn't it be `to == ADDR_DEAD` ?

Reply:
`ADDR_DEAD` is an explicit variable that is defined in the `merge.sol` contract itself. It is basically an edge case. I wrote it in because if someone goes through the process involving `ADDR_DEAD` they will send their token to `ADDR_DEAD,` which then sends to `address(0)`. This updates the collection differently than a merge: the first Transfer is to `ADDR_DEAD` (no change to collection other than a normal `Transfer`). The second step is where we simply see the last `Transfer` event emitted from `ADDR_DEAD` to `address(0)`. 
When one wants to 'burn' a merge NFT, they do so by three methods:
1. The majority of users will do so by simply transferring their NFT to a user that already owns a merge NFT. This in turns carries out:
    - `Transfer(user1, user2, tokenId)`
    - `Transfer(user2, address(0), tokenId)`
2. When someone calls `burn()` they implicitly call `_burnNoEmitTransfer()` which effectively deletes the nft records, respectively. **NOTE: I realized now that this should actually be a separate `call handler()` because `burn()` is not the only time that the `Transfer` event is emitted where. `to` is `address(0)`. This happens in a normal `_merge()` scenario too. So although my synced subgraph showed the same amount of merges as burns, I may be showing inaccurate data.** 
3. When someone sends their NFT directly to `ADDR_DEAD`, which then triggers `_burnNoEmitTransfer()`. `ADDR_DEAD` acts as this odd loophole for people to simply send their nft effectively to `address(0)`. ERC721 doesn't allow someone, by spec design I think (or maybe just suggested implementation), to send to `address(0)` directly. Probably to protect users from doing so accidentally.

### New action items:
- [ ] Create a call handler for `burn()` that increments `collection.totalBurns` and so then we have a summation of b urns that can be used to calculate the "archaic straight burns" that had occurred so far in the collection (if any).